hs_swat_modelinstance/models.py

```diff
@@ -0,0 +1,550 @@
+            raise ValidationError("ModelInput numberOfSubbasins is missing.")
+        if not 'number_of_HRUs' in kwargs:
+            raise ValidationError("ModelInput numberOfHRUs is missing.")
+        if not 'DEM_resolution' in kwargs:
+            raise ValidationError("ModelInput DEMresolution is missing.")
+        if not 'DEM_source_name' in kwargs:
+            raise ValidationError("ModelInput DEMsourceName is missing.")
+        if not 'DEM_source_URL' in kwargs:
+            raise ValidationError("ModelInput DEMsourceURL is missing.")
+        if not 'landUse_data_source_name' in kwargs:
+            raise ValidationError("ModelInput landUseDataSourceName is missing.")
+        if not 'landUse_data_source_URL' in kwargs:
+            raise ValidationError("ModelInput landUseDataSourceURL is missing.")
+        if not 'soil_data_source_name' in kwargs:
+            raise ValidationError("ModelInput soilDataSourceName is missing.")
+        if not 'soil_data_source_URL' in kwargs:
+            raise ValidationError("ModelInput soilDataSourceURL is missing.")
+        metadata_obj = kwargs['content_object']
+        return ModelInput.objects.create(rainfall_time_step=kwargs['rainfall_time_step'],\
+                                             simulation_time_step=kwargs['simulation_time_step'],\
+                                             watershed_area=kwargs['watershed_area'],\
+                                             number_of_subbasins=kwargs['number_of_subbasins'],\
+                                             number_of_HRUs=kwargs['number_of_HRUs'],\
+                                             DEM_resolution=kwargs['DEM_resolution'],\
+                                             DEM_source_name=kwargs['DEM_source_name'],\
+                                             DEM_source_URL=kwargs['DEM_source_URL'],\
+                                             landUse_data_source_name=kwargs['landUse_data_source_name'],\
+                                             landUse_data_source_URL=kwargs['landUse_data_source_URL'],\
+                                             soil_data_source_name=kwargs['soil_data_source_name'],\
+                                             soil_data_source_URL=kwargs['soil_data_source_URL'],\
+                                             content_object=metadata_obj)
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        model_input = ModelInput.objects.get(id=element_id)
+        if model_input:
+            for key, value in kwargs.iteritems():
+                if key in ('rainfall_time_step', 'simulation_time_step', 'watershed_area','number_of_subbasins',\
+                           'number_of_HRUs', 'DEM_resolution', 'DEM_source_name', 'DEM_source_URL',\
+                           'landUse_data_source_name', 'landUse_data_source_URL', 'soil_data_source_name', 'soil_data_source_URL'):
+                    setattr(model_input, key, value)
```

Comment:
so these methods that only update certain keys just ignore other parameters rather than raising an exception? passing invalid parameters should be an error.